src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -28,6 +28,10 @@ internal sealed class PowerShellConfig
         internal static PowerShellConfig Instance => s_instance;
 
         private string psHomeConfigDirectory;
```

Comment:
I suggest we remove `psHomeConfigDirectory` and `appDataConfigDirectory`. Instead, add `psHomeConfigFile` and `appDataConfigFile` and initialize them in `private PowerShellConfig()` (e.g. `Path.Combine(Utils.DefaultPowerShellAppBase, configFileName)`), so we don't need to call `Path.Combine` every time try retrieving a setting value.

In that case, we don't need `psCustomHomeConfigFilePath`, but just set `psHomeConfigFile` in the `SystemConfigFilePath` property. In this way, the method `GetConfigFilePath` will be more simplified.

Reply:
Agreed. In the process, I'll also change the field names. the current names are rather obfuscated considering their usage.

Reply:
Fixed